lib/rack/attack/store_adapters/store_adapter.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+module Rack
+  class Attack
+    module StoreAdapters
+      class StoreAdapter
+        @adapters = []
+
+        class << self
+          def inherited(klass)
+            @adapters << klass
+          end
```

Comment:
Nice!

Will just this `inherited` piece suffice to add support to "add your own cache backend store", by making proxies lookup dynamic instead of hard-coded?

If so, I'll prefer to have that as a small first step, and then re-consider the other changes.

Reply:
Yes.

> If so, I'll prefer to have that as a small first step, and then re-consider the other changes.

You mean extract as a separate PR?

Reply:
> Yes.

Awesome.

> 
> > If so, I'll prefer to have that as a small first step, and then re-consider the other changes.
> 
> You mean extracting as a separate PR?

Yes :-)